### PR TITLE
Make CheckNonce refuse transactions signed by accounts with no providers

### PR DESCRIPTION
**Pull request by zdave-parity**
See #1453.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3700899

**Review by bkchr — APPROVED** [review]
Please add a test.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3876438

**Review by ggwpez — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3884441

File: substrate/client/service/test/src/client/mod.rs [hunk 1]
```diff
@@ -62,22 +61,17 @@ fn construct_block(
...
 
-	let iter = transactions.iter().map(Encode::encode);
-	let extrinsics_root = LayoutV0::<BlakeTwo256>::ordered_trie_root(iter).into();
-
 	let mut header = Header {
 		parent_hash,
 		number,
-		state_root,
-		extrinsics_root,
+		state_root: Default::default(),
```
Comment (zdave-parity) [inline substrate/client/service/test/src/client/mod.rs:71]: AIUI `state_root` and `extrinsics_root` are not used by `Core_initialize_block` and should just be set to 0 here.

File: substrate/client/service/test/src/client/mod.rs [hunk 2]
```diff
@@ -124,19 +118,16 @@ fn construct_block(
...
 	construct_block(
 		backend,
 		1,
 		genesis_hash,
-		array_bytes::hex_n_into_unchecked(
-			"25e5b37074063ab75c889326246640729b40d0c86932edc527bc80db0e04fe5c",
-		),
 		vec![Transfer {
-			from: AccountKeyring::Alice.into(),
+			from: AccountKeyring::One.into(),
```
Comment (zdave-parity) [inline substrate/client/service/test/src/client/mod.rs:130]: Alice has no balance in the genesis state. As this test runtime has no transaction fees I think this was just resulting in the transaction failing. With the `CheckNonce` change the transaction was invalid and causing the runtime to panic.

**Comment by xlc** [comment]
This will make https://github.com/paritytech/polkadot-sdk/issues/1725 harder

**Comment by bkchr** [comment]
> This will make #1725 harder

Good point, but then we need to make it configurable for #1725.

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-3-0/4614/1

**Pull request merged**